src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -418,6 +424,48 @@ public unsafe void SetResponseHeaders()
             }
         }
 
+        public unsafe void SetResponseTrailers()
+        {
+            HttpResponseTrailers.IsReadOnly = true;
+            foreach (var headerPair in HttpResponseTrailers)
+            {
+                var headerValues = headerPair.Value;
+
+                if (headerValues.Count == 0)
+                {
+                    continue;
+                }
+
+                string headerValue;
+                if (headerValues.Count == 1)
+                {
+                    headerValue = headerValues[0];
+                }
+                else
+                {
+                    var builder = new StringBuilder();
+                    builder.Append(headerValues[0]);
+                    for (var i = 1; i < headerValues.Count; i++)
+                    {
+                        builder.Append(',');
+                        builder.Append(headerValues[i]);
+                    }
+
+                    headerValue = builder.ToString();
+                }
+
+                var headerValueBytes = Encoding.UTF8.GetBytes(headerValue);
+                fixed (byte* pHeaderValue = headerValueBytes)
+                {
+                    var headerNameBytes = Encoding.UTF8.GetBytes(headerPair.Key);
```

Comment:
Names should be ASCII.

Reply:
I didn't even know we allowed response header values to be UTF8 in in-proc. Kestrel doesn't.